install/resources/upgrade51.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * MyBB 1.8
+ * Copyright 2014 MyBB Group, All Rights Reserved
+ *
+ * Website: http://www.mybb.com
+ * License: http://www.mybb.com/about/license
+ *
+ */
+
+/**
+ * Upgrade Script: 1.8.24
+ */
+
+$upgrade_detail = array(
+    "revert_all_templates" => 0,
+    "revert_all_themes" => 0,
+    "revert_all_settings" => 0
+);
+
+@set_time_limit(0);
+
+function upgrade51_dbchanges()
+{
+    global $output, $cache, $db, $mybb;
+
+    $output->print_header("Updating Database");
+
+    echo "<p>Performing necessary upgrade queries...</p>";
+    flush();
+
+    // Add new setting for new usergroup permission if group members can hide online status
+    $db->add_column("usergroups", "canbeinvisible", "smallint NOT NULL default '1' AFTER canusercp");
```

Comment:
This will need a switch for mysql,sqlite/postgresql (tinyint vs smallint)

For example:
https://github.com/mybb/mybb/blob/bdd1f65378685f164c2b47ec2adc81b807bdc454/install/resources/upgrade43.php#L48-L56

Reply:
Note: MariaDB supports the `BOOLEAN` column type. This is a synonym for `TINYINT(1)`. Not sure about MySQL.

Reply:
Seems MySQL also accepts tinyint(1) so i've pushed the upgrade changes.